### ISSUE TITLE
Complete `isolated` function support for function type descriptors and anonymous function expressions

### ISSUE THREAD

**Issue by MaryamZi** [ballerina-platform/ballerina-lang#25565]
**Description:**
https://github.com/ballerina-platform/ballerina-lang/pull/25423 adds initial support for isolated functions. Need to $title.

Need to also verify support for scenarios like

```
type F isolated function (int) returns int;
F f = x => x + 1;
```

Depends on https://github.com/ballerina-platform/ballerina-lang/issues/25563.

**Comment by MaryamZi** [comment]
Once this is supported, need to mark langlib methods as isolated too.

**Comment by MaryamZi** [comment]
Need to also verify https://github.com/ballerina-platform/ballerina-spec/issues/145#issuecomment-668404584.

**Comment by MaryamZi** [comment]
`isolated` function support for function type descriptors and anonymous function expressions was merged with #25761. The langlib functions that are always `isolated` were also marked as `isolated` in line with the spec.

Closing this issue since the pending tasks have their own issues
- Analyze closure isolated-ness wrt to the stack frame #25918 
- Introduce the `@isolatedParam` annotation and related analysis #25792
